mmtk/src/object_model.rs

```diff
@@ -48,9 +48,9 @@ impl ObjectModel<OpenJDK> for VMObjectModel {
                 unsafe { (dst + i).store((src + i).load::<u8>()) };
             }
         }
-        let start = Self::ref_to_object_start(to);
+        let start = <Self as ObjectModel<OpenJDK<COMPRESSED>>>::ref_to_object_start(to);
```

Comment:
The `<Self as .....>` is necessary because the `VMObjectModel` type actually implements both `ObjectModel<OpenJDK<true>>` and `ObjectModel<OpenJDK<false>>`.

This can be made unnecessary by giving `struct VMObjectModel` a const parameter:

```rust
pub struct VMObjectModel<const COMPRESSED: bool> {}
```

and let it only implement the trait with the same const argument:

```rust
impl<const COMPRESSED: bool> ObjectModel<OpenJDK<COMPRESSED>> for VMObjectModel<COMPRESSED> {
```

Then `VMObjectModel<true>` only implements `ObjectModel<OpenJDK<true>>`, and `VMObjectModel<false>` only implements `ObjectModel<OpenJDK<false>>`.

But the current code `<Self as ObjectModel<OpenJDK<COMPRESSED>>>` is still OK given that it does not appear too often.

Reply:
Fixed